src/lib/mina.ts

```diff
@@ -301,12 +301,21 @@ function LocalBlockchain({
       txn.sign();
 
       let partiesJson = partiesToJson(txn.transaction);
-
-      ledger.applyJsonTransaction(
-        JSON.stringify(partiesJson),
-        String(accountCreationFee),
-        JSON.stringify(networkState)
-      );
+      try {
+        ledger.applyJsonTransaction(
+          JSON.stringify(partiesJson),
+          String(accountCreationFee),
+          JSON.stringify(networkState)
+        );
+      } catch (err: any) {
+        try {
+          // reverse errors so they match order of account updates
```

Comment:
I've been thinking that instead of just throwing error messages, we could catch them and match them up with the string descriptions that live here: https://github.com/MinaProtocol/mina/blob/91acb8f454111bc65ec85af9888c73217951efd2/src/lib/mina_base/transaction_status.ml#L324

Would this try block be the place we would do it? If so, what's the best way of generating these strings without manually copying them so we can instead throw the messages instead? Would we just expose these in the SnarkyJS bindings and have a dictionary we could work with?

Reply:
yeah it seems we could just import them to `snarky_js_bindings_lib.ml` and replace the statuses with strings there -- that should be fairly easy!